editoast/src/views/v2/train_schedule/projection.rs

```diff
@@ -0,0 +1,599 @@
+    track_section_ranges: Vec<TrackRange>,
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize, ToSchema)]
+struct SpaceTimeCurve {
+    // List of positions of a train in mm
+    // Both positions and times must have the same length
+    #[schema(min_items = 2)]
+    positions: Vec<u64>,
+    // List of times in ms since `departure_time` associated to a position
+    #[schema(min_items = 2)]
+    times: Vec<u64>,
+}
+
+/// Project path output is described by time-space points and blocks
+#[derive(Debug, Deserialize, Serialize, ToSchema)]
+struct ProjectPathTrainResult {
+    /// Departure time of the train
+    departure_time: DateTime<Utc>,
+    /// Rolling stock length in mm
+    rolling_stock_length: u64,
+    #[serde(flatten)]
+    #[schema(inline)]
+    cached: CachedProjectPathTrainResult,
+}
+
+/// Project path output is described by time-space points and blocks
+#[derive(Debug, Deserialize, Serialize, ToSchema)]
+struct CachedProjectPathTrainResult {
+    /// List of space-time curves sections along the path
+    #[schema(inline)]
+    space_time_curves: Vec<SpaceTimeCurve>,
+    /// List of signal updates along the path
+    #[schema(inline)]
+    signal_updates: Vec<SignalUpdate>,
+}
+
+/// Projects the space time curves and paths of a number of train schedules onto a given path
+///
+/// - Returns 404 if the infra or any of the train schedules are not found
+/// - Returns 200 with a hashmap of train_id to ProjectPathTrainResult
+///
+/// Train schedules that are invalid (pathfinding or simulation failed) are not included in the result
+#[utoipa::path(
+    tag = "train_schedulev2",
+    params(
+        ("infra" = i64, Query, description = "The infra id"),
+        ("ids" = Vec<i64>, Query, description = "Ids of train schedule"),
+    ),
+    responses(
+        (status = 200, description = "Project Path Output", body = HashMap<i64, ProjectPathTrainResult>),
+    ),
+)]
+#[post("/project_path")]
+async fn project_path(
+    db_pool: Data<DbPool>,
+    redis_client: Data<RedisClient>,
+    core_client: Data<CoreClient>,
+    params: QsQuery<SimulationBatchParams>,
+    data: Json<ProjectPathInput>,
+) -> Result<Json<HashMap<i64, ProjectPathTrainResult>>> {
+    let ProjectPathInput {
+        track_section_ranges: path_track_ranges,
+        ..
+    } = data.into_inner();
+    let path_projection = PathProjection::new(&path_track_ranges);
+    let query_props = params.into_inner();
+    let train_ids = query_props.ids;
+    let infra_id = query_props.infra;
+    let conn = &mut db_pool.clone().get().await?;
+    let mut redis_conn = redis_client.get_connection().await?;
+
+    let infra = Infra::retrieve_or_fail(conn, infra_id, || TrainScheduleError::InfraNotFound {
+        infra_id,
+    })
+    .await?;
+
+    let train_schedule_batch: Vec<TrainSchedule> =
+        TrainSchedule::retrieve_batch_or_fail(conn, train_ids, |missing| {
+            TrainScheduleError::BatchTrainScheduleNotFound {
+                number: missing.len(),
```

Comment:
```suggestion
                missing,
```

for consistency

Reply:
Would you like us to return the set of all missing ids ?

Reply:
yep, as a rest user I'd like to know which of my ids were incorrect

Reply:
Sadly, `delete_batch_or_fail` return the number of failures and not the HashSet 😞